torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -463,19 +545,16 @@ std::tuple<std::vector<PartitionDesc>,std::vector<PartitionDesc>,bool> emitCompi
   {
     size_t input_index = 0;
     for(auto p : subgraph.inputs()) {
-      if (auto chunk = usedInFusedChunk(p)) {
+      if (Node * chunk = usedInFusedChunk(p)) {
         int64_t dim = chunk->i(attr::dim);
         int64_t chunks = chunk->i(attr::chunks);
-				auto tensor_type = p->type()->cast<TensorType>();
-        chunk_desc.emplace_back(tensor_type, chunks, dim, false);
-
+        chunk_desc.emplace_back(agraph.input_desc[input_index++], chunks, dim);
         for (auto * o : chunk->outputs()) {
           flat_inputs.emplace_back(o, *chunk_desc.back().subtensorDesc);
         }
-        ++input_index;
       } else {
-        flat_inputs.emplace_back(p, agraph.input_desc[input_index++]);
         chunk_desc.emplace_back();
+        flat_inputs.emplace_back(p, agraph.input_desc[input_index++]);
```

Comment:
If `input_index++` is always incremented, why not just move it out of the conditional XD

Reply:
Idk, doesn't seem to be significantly improve the code if it was to be moved out of the branches.